app/views/properties/show.html.erb

```diff
@@ -90,24 +90,34 @@
         </h5>
         <hr>
         <div class="row small-up-2">
-            <%= Amenity.find_each do |amenity| %>
-            <% if @property.amenities.include?(amenity) %>
-            <div class="column">
-                <label>
-                    <%= amenity.name %>
-                </label>
-              </div>
-              <% else %>
-              <div class="column">
-                  <label>
-                      <del><%= amenity.name %></del>
-                  </label>
+            <div class="js-amenities">
+                <% Amenity
+                    .find_each
+                    .each do |amenity|
+                %>
+                <% if @property.amenities.include?(amenity) %>
+                <div class="column <%= amenity.icon.size == 0 ? 'js-amenityHide' : ''%>">
+                    <label>
+                        <% if amenity.icon.size > 0 %>
```

Comment:
In Ruby, objects if not nil evaluate to true. so you can just say `<% if amenity.icon %>`. If you want to be more explicit, it is also appropriate to use `<% if amenity.icon.present? %>` which checks if the object is nil and if it is empty but is rails specific.